src/test/java/org/jenkinsci/plugins/github_branch_source/EventsTest.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package org.jenkinsci.plugins.github_branch_source;
+
+import jenkins.scm.api.SCMEvent;
+import jenkins.scm.api.SCMHeadEvent;
+import jenkins.scm.api.SCMSourceEvent;
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.plugins.github.extension.GHSubscriberEvent;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class EventsTest {
+    /**
+     * All tests in this class only use Jenkins for the extensions
+     */
+    @ClassRule
+    public static JenkinsRule r = new JenkinsRule();
+
+    private static SCMEvent.Type firedEventType;
+    private static GHSubscriberEvent ghEvent;
+
+    @Test
+    public void pushEvents() throws Exception {
```

Comment:
you are using a `@ClassRule` please split each test into individual test cases rather than piling on a load of scenarios in one test. Also name each test with what it is testing and what the outcome should be